etna/models/statsforecast.py

```diff
@@ -0,0 +1,795 @@
+
+        return grid
+
+
+class StatsForecastAutoThetaModel(
+    PerSegmentModelMixin, PredictionIntervalContextIgnorantModelMixin, PredictionIntervalContextIgnorantAbstractModel
+):
+    """
+    Class for holding :py:class:`statsforecast.models.AutoTheta`.
+
+    Method ``forecast`` only works on ouf-of-sample data that goes right after training data.
+
+    Method ``predict`` only works on in-sample data.
+    """
+
+    def __init__(
+        self,
+        season_length: int = 1,
+        decomposition_type: str = "multiplicative",
+        model: Optional[str] = None,
+    ):
+        """Init model with given params.
+
+        Parameters
+        ----------
+        season_length:
+            Number of observations per unit of time. Ex: 24 Hourly data.
+        decomposition_type:
+            Sesonal decomposition type, 'multiplicative' (default) or 'additive'.
+        model:
+            Controlling Theta Model. By default searches the best model.
+        """
+        self.season_length = season_length
+        self.decomposition_type = decomposition_type
+        self.model = model
+        super().__init__(
+            base_model=_AutoThetaAdapter(
+                season_length=self.season_length, decomposition_type=self.decomposition_type, model=self.model
+            ),
+        )
+
+
+class StatsForecastAutoCESModel(
+    PerSegmentModelMixin,
+    NonPredictionIntervalContextIgnorantModelMixin,
+    NonPredictionIntervalContextIgnorantAbstractModel,
+):
+    """
+    Class for holding :py:class:`statsforecast.models.AutoCES`.
+
+    Method ``forecast`` only works on ouf-of-sample data that goes right after training data.
+
+    Method ``predict`` only works on in-sample data.
+    """
+
+    def __init__(self, season_length: int = 1, model: str = "Z"):
+        """Init StatsForecastAutoCESModel with given params.
+
+        Parameters
+        ----------
+        season_length:
+            Number of observations per unit of time. Ex: 24 Hourly data.
+        model:
+            Controlling state-space-equations.
+        """
+        self.season_length = season_length
+        self.model = model
+        super().__init__(
+            base_model=_AutoCESAdapter(season_length=self.season_length, model=self.model),
+        )
+
+
+class StatsForecastAutoETSModel(
+    PerSegmentModelMixin, PredictionIntervalContextIgnorantModelMixin, PredictionIntervalContextIgnorantAbstractModel
+):
+    """
+    Class for holding :py:class:`statsforecast.models.AutoETS`.
+
+    Method ``forecast`` only works on ouf-of-sample data that goes right after training data.
+
+    Method ``predict`` only works on in-sample data.
+    """
+
+    def __init__(self, season_length: int = 1, model: str = "ZZZ", damped: Optional[bool] = None):
+        """Init model with given params.
+
+
+        Parameters
+        ----------
+        season_length:
+            Number of observations per unit of time. Ex: 24 Hourly data.
+        model:
```

Comment:
Here I used the type from `statsforecast` library. The possible values are reflected in their [documentation](https://nixtla.github.io/statsforecast/src/core/models.html#autoets-1). It can be potentially expressed as `Literal`, but it will take a lot of values (48).